cylc/flow/pathutil.py

```diff
@@ -297,15 +298,27 @@ def remove_dir_and_target(path: Union[Path, str]) -> None:
                 "Removing symlink and its target directory: "
                 f"{path} -> {target}"
             )
-            rmtree(target, onerror=handle_rmtree_err)
+            _rmtree(target)
         else:
             LOG.info(f'Removing broken symlink: {path}')
         os.remove(path)
     elif not os.path.exists(path):
         raise FileNotFoundError(path)
     else:
         LOG.info(f'Removing directory: {path}')
-        rmtree(path, onerror=handle_rmtree_err)
+        _rmtree(path)
+
+
+def _rmtree(target, retries=5, sleep_time=0.5):
+    for _try_num in range(retries):
+        try:
+            rmtree(target)
+            return
+        except OSError as exc:
```

Comment:
To avoid retrying the `shutil.rmdirs` when it is not appropriate to do so we should filter for "directory not empty" errors. The `errno` module is the tool for the job (error numbers are OS dependent):

```python
        except OSError as exc:
            if exc.errno == errno.ENOTEMPTY:
                sleep(sleep_time)
            else:
                raise
```

https://docs.python.org/3/library/errno.html?highlight=errno#errno.ENOTEMPTY

Reply:
That's how `handle_rmtree_err` works

Reply:
Why the decision to retry 5 times? (There is no docstring for this function)